### PR TITLE
feat: port to app-platform

### PR DESCRIPTION
**Pull request by HendrikThePendric**
This PR ports this app to the app-platform. The checks are still failing, which is because of lint issues. These have been addressed in #206. I've done this is a separate PR so the current PR has a clear focus.

I'd like to propose the following steps:
1. Review this PR and fix things until approved
2. Review #206 and fix things until approved
3. Merge #206 into this one
4. Then the checks should all become green, if not there's work to do
5. Merge this into master

### CODE REVIEW

File: .huskyrc.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+const { config } = require('@dhis2/cli-style')
+const husky = require(config.husky)
+
+const tasks = arr => arr.join(' && ')
+
+module.exports = {
+    hooks: {
+        ...husky.hooks,
+        'pre-commit': tasks([
+            'd2-style js check --staged',
```
Comment (ismay) [inline .huskyrc.js:10]: Should we destructure `...husky.hooks['pre-commit']` here as well? I don't know if it's the intention that we overwrite `pre-commit` or merge with the ones supplied by the dhis2 config.

Comment (HendrikThePendric) [reply .huskyrc.js:10]: This file was generated by running the command `npx d2-style install project/react --force`. So if you would like a change here, you should go via that repo.

More info:
- [This](https://github.com/dhis2/cli-style/blob/master/config/husky-frontend.local.js) is where the file comes from 
- All files in this commit 4211085abc6db401ed6a1a2a14a2145986e95d71 were auto-generated, apart from the change to `.browserslistrc`

File: src/components/App/AppWrapper.jsx [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+        d2Config: {
+            schemas: ['messageConversation'],
+            // This app prefers doing unversioned API calls
+            baseUrl: `${baseUrl}/api`,
+        },
+    })
+
+    if (!d2) {
+        return (
+            <AddD2Context d2={d2}>
```
Comment (ismay) [inline src/components/App/AppWrapper.jsx:26]: It is unclear to me why we pass `d2` here if d2 is falsey?

File: .huskyrc.js (hunk 1 above)
Comment (ismay) [reply .huskyrc.js:10]: Ah yeah of course. In that case I assume this is fine 👍

**Comment by ismay** [comment]
Looks good 👍. Went through the app (with debug/2.35.0) and everything works as expected. Hendrik explained that the messageConversationId could be omitted, similar to what we did earlier with the user-app. So only comment remaining is the question about the falsey d2.

**Comment by ismay** [comment]
Ah. One other thing. I believe the .env files can all be removed. If I'm not mistaken the env vars for production builds should be supplied by the build environment, and during development they're no longer needed either (since we have the login dialog).

**Review by ismay — APPROVED** [review]
Looks good. Tested the dev and production build, and both work as expected without the .envs.

**Pull request merged**